lib/srv/db/server.go

```diff
@@ -223,8 +223,11 @@ func (s *Server) initHeartbeat(ctx context.Context, server types.DatabaseServer)
 
 func (s *Server) getServerInfoFunc(server types.DatabaseServer) func() (services.Resource, error) {
 	return func() (services.Resource, error) {
-		s.mu.Lock()
-		defer s.mu.Unlock()
+		// Make sure to return a new object, because it gets cached by
+		// heartbeat and will always compare as equal otherwise.
+		s.mu.RLock()
+		server = server.Copy()
+		s.mu.RUnlock()
```

Comment:
Why not move the lock to `Copy`?

Reply:
@russjones Copy is the method on the resource, I don't think we want to lock the resource's method?

Reply:
Missed that.